sklearn/decomposition/online_lda.py

```diff
@@ -0,0 +1,653 @@
+                np.dot(cnts / norm_phi, exp_topic_word_d.T)
+            exp_doc_topic_d = np.exp(_dirichlet_expectation(doc_topic_d))
+            norm_phi = np.dot(exp_doc_topic_d, exp_topic_word_d) + 1e-100
+
+            meanchange = mean_change(last_d, doc_topic_d)
+            if meanchange < mean_change_tol:
+                break
+        doc_topic_distr[d, :] = doc_topic_d
+
+        # Contribution of document d to the expected sufficient
+        # statistics for the M step.
+        if cal_diff:
+            component_diff[:, ids] += np.outer(exp_doc_topic_d, cnts / norm_phi)
+
+    return (doc_topic_distr, component_diff)
+
+
+class LatentDirichletAllocation(BaseEstimator, TransformerMixin):
+
+    """
+    Online Latent Dirichlet Allocation implementation with variational inference
+
+    References
+    ----------
+    [1] "Online Learning for Latent Dirichlet Allocation", Matthew D. Hoffman, 
+        David M. Blei, Francis Bach
+
+    [2] Matthew D. Hoffman's onlineldavb code. Link:
+        http://www.cs.princeton.edu/~mdhoffma/code/onlineldavb.tar
+
+
+    Parameters
+    ----------
+    n_topics : int, optional (default: 10)
+        Number of topics.
+
+    doc_topic_prior : float, optional (defalut: 0.1)
+        Prior of document topic distribution `theta`. In general, it is `1 / n_topics`.
+        In literature, it is Hyperparameter parameter `alpha`.
+
+    topic_word_prior : float, optional (default: 0.1)
+        Prior of topic word distribution `beta`. In general, it is `1 / n_topics`.
+        In literature, it is hyperparameter parameter `eta`.
+
+    learning_method : 'batch' | 'online', default: 'online'
+        Method used to update `_component`. Only used in `fit` method.
+        In general, if the data size is large, online update will be much faster than
+        batch udpate.
+        Valid options::
+
+            'batch': Batch variational bayes method. Use all training data in each EM udpate.
+                Old `components_` will be overrided in each iteration.
+            'online': Online variational bayes method. In each EM update, use min batch of
+                training data to update `components_` variable incrementally. The weight in
+                each udpate is controlled by `learning_decay` and `learning_offset` parameter
+
+    learning_decay : float, optional (default: 0.7)
+        It is a parameter that control learning weight for `_component` in online learning.
+        The value should be set between (0.5, 1.0] to guarantee asymptotic convergence.
+        When the value is 0.0 and batch_size is `n_samples`, the udpate is same as batch learning.
+        In literature, it is parameter `kappa`.
+
+    learning_offset : float, optional (default: 1000.)
+        A (positive) parameter that downweights early iterations in online learning.
+        It should be greater than 1.0. In literature, it is parameter `tau0`.        
```

Comment:
I think the formulation should be "In the literature, this is called tau0" [also in other places]